### DISCUSSION TITLE
How to calculate the z value for getColor?

### DISCUSSION THREAD

**Discussion by danvoyce** [visgl/deck.gl#7969] [Q&A]
I'm trying to replicate the [line example](https://github.com/visgl/deck.gl/blob/8.9-release/examples/website/line/app.jsx#L29), however I need to get my data in the correct format in order for the colors to make sense.

Looking at the data in the example it shows:

```
{
  "start":[8.714, 54.2638, 11582.4], 
  "end":[8.6875, 54.2501, 11582.4],
  "name":"SAS52H"
}
```

The last item in each array. ie the `z` value is used to generate the color. I assume this has been calculated somehow based on the number of overlaps. Can anyone steer me in the direction of how this could be calculated?

cc @ibgreen I believe you made changes to this line of code last so maybe you can help?

Many thanks 🙏

**Comment by ibgreen** [comment]
To me it looks like the color is simply based on the elevation in meters, with red color component being emphasized at low altitudes. 

Given the row data you gave, `row.start[2]` = 11582.4:
```
function getColor(row) {
  const z = row.start[2];
  const r = z / 10000;

  return [255 * (1 - r * 2), 128 * r, 255 * r, 255 * (1 - r)];
}
```

**Reply by danvoyce** [reply]
Thank you @ibgreen for such a speedy response! 

Yes of course, that makes complete sense. I was thinking it was something more clever than that. I'm trying to visualise vessel traffic in a heatmap-esq fashion. Therefor need to set the color according to the denser areas. Simply relying on the alpha channel and stacking them doesn't quite give the same effect.

**Reply by KiwiKilian** [reply]
@danvoyce did you find a suitable solution? Facing a similar problem.
